common/src/main/scala/hydra/common/alerting/NotificationRequestBaker.scala

```diff
@@ -0,0 +1,40 @@
+package hydra.common.alerting
+
+import cats.Monad
+import cats.syntax.all._
+import eu.timepit.refined.types.string.NonEmptyString
+import hydra.common.alerting.AlertProtocol._
+import spray.json.JsonWriter
+
+import java.net.InetAddress
+import scala.language.higherKinds
+
+trait NotificationRequestBaker[F[_], T] {
+
+  def bake[K: JsonWriter](notificationInfo: NotificationScope,
+                          notificationMessage: NotificationMessage[K])
+                         (source: T): F[Option[NotificationRequest]]
+}
+
+object NotificationRequestBaker {
+
+  implicit def basicNotificationBacker[F[_]: Monad]: NotificationRequestBaker[F, Option[NonEmptyString]] =
+    new NotificationRequestBaker[F, Option[NonEmptyString]] {
```

Comment:
Why it is side effect? I don`t think that this is a side effect

Reply:
@Ol-nikulin Actually someone says it’s not but I saw most arguments that creating objects it’s side effect because  every object changes “world”. There are many nuances and all depends on memory allocations and live time.

But good practice is to make classes via factory method and good practice is to make this factory methods as pure.

Often used in practice a pattern like pure interfaces:

trait SomeClass`[F[_]]` {

}

object SomeClass {

def make`[F[_]]`(): F[SomeClass[F]] = {

…..

}
}

Reply:
discussed on the meeting (14.07.2022)